### PR TITLE
CUMULUS-1413 Alternative test fix.

### PR DESCRIPTION
**Pull request by flamingbear**
Instead of using the model to delete the granule.  Have the buildDatabaseFiles
read the source + host when the sns2elasticsearch creates the granules.

This should create files that will at least have a key and bucket when run
through aws.fileExists.

----

#

**Summary:**

Addresses [CUMULUS-1413: prevent DB clutter when testing](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1413)

## Changes

* Adds last ditch step to buildDatabaseFiles that will look in the source key and parse an s3 url for bucket/key

## PR Checklist

- [ N/A ] Update CHANGELOG
- [ X ] Unit tests
- [ X ] Adhoc testing
- [ X ] Integration tests

### CODE REVIEW

File: packages/api/tests/lib/FileUtils/test-parseSource.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  };
+
+  const expected = { ...file };
+
+  const actual = parseSource(file);
+
+  t.deepEqual(expected, actual);
+});
+
+test('leaves file unchanged if key is not null', (t) => {
```
Comment (laurenfrederick) [inline packages/api/tests/lib/FileUtils/test-parseSource.js]: The key is null here right? Same for line 38.

Comment (flamingbear) [reply packages/api/tests/lib/FileUtils/test-parseSource.js]: In this case it's actually undefined, not null.  I could add another test to show one where, key has a value but bucket is null.    The getBucket and getKey functions return explicit null values, and so the file object has key: null and bucket: null, when the normal mode of operation doesn't find what it's looking for.  
I did it this way to show that leaving a value off the file, is not actually null.  I clarified by adding a test without a bucket or key and adding in not null values to these two,.

Comment (flamingbear) [reply packages/api/tests/lib/FileUtils/test-parseSource.js]: 7151f5e96
